index.js

```diff
@@ -94,7 +97,36 @@ function parseChar(character, state) {
     } else {
       state.escaped = false;
     }
-  } else if (state.regexp) {
+    return end();
+  } else if (state.backTickDepth) {
+    if (character === '$' && !state.escaped) {
+      state.dollarSign = true;
+      return end();
+    } else if (character === '{' && state.dollarSign && !state.escaped) {
```

Comment:
Is escaped relevant here?  I would guess you're trying to catch `` `$\{` `` but that should already end up with `dollarSign` set to `false`.